flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -60,12 +60,18 @@ bool PrescanAndParseAction::BeginSourceFileAction() {
 }
 
 bool PrescanAndSemaAction::BeginSourceFileAction() {
-  return RunPrescan() && RunParse() && RunSemanticChecks();
+  return RunPrescan() && RunParse() && RunSemanticChecks() &&
+      GenerateRtTypeTables();
 }
 
 bool PrescanAndSemaDebugAction::BeginSourceFileAction() {
-  // Semantic checks are made to succeed unconditionally.
-  return RunPrescan() && RunParse() && (RunSemanticChecks() || true);
+  // This is a "debug" action for development purposes. To facilitate this, the
+  // semantic checks are made to succeed unconditionally to prevent this action
+  // from exiting early (i.e. in the presence of semantic errors). We should
+  // never do this in actions intended for end-users or otherwise regular
+  // compiler workflows!
+  return RunPrescan() && RunParse() && (RunSemanticChecks() || true) &&
+      (GenerateRtTypeTables() || true);
```

Comment:
`|| true`? Why be cryptic?
```suggestion
if (RunPrescan() && RunParse()) {
  (void)RunSemanticChecks();
  (void)GenerateRtTypeTables();
}
```

Reply:
It's more for consistency with other actions than anything else. It's either (from [PrescanAndSemaAction::BeginSourceFileAction](https://github.com/llvm/llvm-project/blob/main/flang/lib/Frontend/FrontendActions.cpp#L49-L52)):
```cpp
  return RunPrescan() && RunParse() && RunSemanticChecks() && GenerateRtTypeTables();
```
or (from [PrescanAndSemaDebugAction::BeginSourceFileAction](https://github.com/llvm/llvm-project/blob/main/flang/lib/Frontend/FrontendActions.cpp#L54-L62)):
```cpp
return RunPrescan() && RunParse() && (RunSemanticChecks() || true) && (GenerateRtTypeTables() || true);
```
This way, IMHO, it's easier to see the difference between the two.

Reply:
It's unnecessarily cryptic. Coders see `if` statements all the time.

Reply:
Done